### ISSUE TITLE
steam write to file

### ISSUE THREAD

**Issue by ISPLab** [ISPLab/P8#146]
помните мы с вами в феврале общались по потоковой записи в файл, и даже метод создавали, вы его реализовали?

**Comment by kaa1965** [comment]
Нет, он ещё не реализован. Недавно с Булавским и с Разработчиками Ровера только обсудили что и как в него должно писаться.

**Comment by ISPLab** [comment]
слушайте а байты потоково пишутся по разному в зависимости от контента в них?

**Comment by kaa1965** [comment]
Вариант 1:
//fileName - полный путь и имя файла
public static void SaveFile(string fileName, byte[] bytes)
        {
            using (System.IO.FileStream fs = new FileStream(fileName, FileMode.Create, FileAccess.Write, FileShare.None))
            {                
                fs.Write(bytes, 0, bytes.Length);
            }
        }

Вариант 2:
- открываешь 
fs = new FileStream(fileName, FileMode.Create, FileAccess.Write, FileShare.None);
- пишешь порцию:
fs.Write(bytes, 0, bytes.Length);
fs.Flush();//скидываешь на диск, типа чтобы данные не повисли в буферах и памяти FileStream-а то когда оно окажется на диске одному богу известно...
- пишешь следующие порции бинарных данных
- в конце работы завершаешь работу:
fs.Close();
fs.Dispose();
Примечание: параметры при открытии FileStream-а могут быть и другими. Указанные означают, что файл всё время создаётся новый, если файл уже существует, то он будет перезаписан (FileMode.Create) + кто то другой работать с этим файлом не сможет ни на запись ни на чтение (FileShare.None).
